src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -142,28 +142,45 @@ private void ResetSelection()
 			}
 		}
 
-		internal override void OnSelectorItemIsSelectedChanged(SelectorItem container, bool oldIsSelected, bool newIsSelected)
+		internal override void ChangeSelectedItem(object item, bool oldIsSelected, bool newIsSelected)
 		{
 			if (!_modifyingSelectionInternally)
 			{
-				var item = ItemFromContainer(container);
 
-				if (!newIsSelected)
+				//Handle selection
+				switch (SelectionMode)
 				{
-					SelectedItems.Remove(item);
-				}
-				else
-				{
-					if (!SelectedItems.Contains(item))
-					{
-						if (!IsSelectionMultiple
-							&& SelectedItems.FirstOrDefault() is object selectedItem)
+					case ListViewSelectionMode.None:
+						break;
+					case ListViewSelectionMode.Single:
+						if (!newIsSelected)
+						{
+							SelectedIndex = -1;
+						} else
 						{
-							SelectedItems.Remove(selectedItem);
+							SelectedIndex = IndexFromItem(item);
 						}
-
-						SelectedItems.Add(item);
-					}
+						break;
+					case ListViewSelectionMode.Multiple:
+					case ListViewSelectionMode.Extended:
+						if (!newIsSelected)
+						{
+							SelectedItems.Remove(item);
+						}
+						else
+						{
+							if (!SelectedItems.Contains(item))
+							{
+								if (!IsSelectionMultiple
```

Comment:
In retrospect, I may need to revise this and add the functionality back. As it is this will never actually hit.